### DISCUSSION TITLE
Intermediate mode?

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [picoe/Eto#2063] [Ideas]
What about a intermediate mode wrapper?

Its really easy to build up complex layouts with intermediate mode GUIs
```cs
ImEto.BeginForm("MyForm", 640, 480);
   ImEto.BeginHorizontal();
     ImEto.BeginVertical();
        ImEto.Text("Some text", AlignmentX.Center, AlignmentY.Center);
     ImEto.EndVertical();
     ImEto.BeginGroup();
       if (ImEto.TextButton("my text button"), 4,4,4,4); // the 4's are padding around the control.
       {
           //I was pressed.
       }
     ImEto.EndGroup();
   ImEto.EndHorizontal();
ImEto.EndForm();
```

It might be better also for gamedev since the gui loop would take care of itself.

**Comment by pha3z** [comment]
I believe you are talking about "Immediate mode."   Immediate Mode UIs render their entire frame contents with every loop of the application.  Hence they're typically found in GPU-based contexts such as games and simulations.  Obviously you can do an immediate mode UI for a regular application, but its not necessarily the best choice.

Also, I don't personally agree with the philosophy that immediate mode UIs are automatically easier to layout.   The  way in which you code the UI and the way its constructed are not specific to whether or not it runs as Immediate Mode.  I've seen other people claim the contrary, but I think that's because they're only seeing a common "pattern of use".    Immediate mode does not have to imply a specific way of coding layout.
